### ISSUE TITLE
Docstring linting fails if an encoding declaration is at the top of the source file

### ISSUE THREAD

**Issue by rafaelbco** [DamnWidget/anaconda#686]
## Expected Behaviour
I expect the docstring linting to work on files with any valid encoding declaration. Per [PEP263](https://www.python.org/dev/peps/pep-0263/) the following line is a valid declaration. In fact, it's the first example mentioned in the PEP.

```
# coding=utf8
```

## Actual Behaviour
If I put the aforementioned declaration in the first line in the top of a Python file, the docstring linting does not work. 

## Steps to Reproduce
Create this file:

```
#  coding=utf8


def f():
    '''a'''
    print(1)
```

Confirm docstring linting is not working by using the command "Anaconda: Show error list".

Change the first line to:

```
# coding = utf8
```

Then issue the same command and 5 docstring errors will show up.

## ST3, Anaconda and OS versions
- ST3: Build 3126
- Anaconda: v2.1.20 and also commit 2c33c8e63a484feb8620c7f9bd04913c12d58bee and also 
- Linux Mint 17.3
- Python 2.7.12

## ST3 Console Logs
Nothing.

## Anaconda's JsonServer Logs
**Note**: Anaconda's JsonServer logs can be found in:
* Linux: ~/.local/share/anaconda/logs
* OS X: ~/Library/Logs/anaconda
* Windows: %APPDATA\\Anaconda\\Logs

Could not find the log.

**Comment by rafaelbco** [comment]
Here's what I've found. Keep in mind I'm on Python 2.7.

When I run `pydocstyle` from the command line (the same version included in Anaconda), the problem does not occurs.

Digging in the code, I found out Anaconda use the method `checker.ConventionChecker.check_source` as the entry point to `pydocstyle`. However it passes an `unicode` object as the first argument. When calling `pydocstyle` from the command line it passes an `str`. Fast forward, the source code ends up being processed by the `parser.Parser.parse` method. And it uses the builtin `compile` to check if it's valid Python code. 

This is where things get interesting: `compile` raises an error if you pass an `unicode` with an encoding declaration. But with one caveat: it does not detects encoding declarations properly. I wrote this program:

```
p1 = """# coding = utf8\nprint 1"""
p2 = """# coding=utf8\nprint 1"""

programs = [p1, p2]
programs.extend(p.decode('utf8') for p in list(programs))

for p in programs:
    try:
        print 'Compiling program {}'.format(repr(p))
        compile(p, 'p.py', 'exec')
        print 'Ok!'
    except SyntaxError as e:
        print 'Error: {}'.format(e)
```

And it gives the following output:

```
Compiling program '# coding = utf8\nprint 1'
Ok!
Compiling program '# coding=utf8\nprint 1'
Ok!
Compiling program u'# coding = utf8\nprint 1'
Ok!
Compiling program u'# coding=utf8\nprint 1'
Error: encoding declaration in Unicode string (p.py, line 0)
```

To be consistent both the `unicode` program strings should have raised an error. It would make our problem worse though, since more styles of encoding declarations would cause problems.

So, to fix the problem we have two options: strip encoding declaration before feeding it in to `pydocstyle` or encode the `unicode` containg the source code, transforming it to `str`. 

I prefer the second approach, since `pydocstyle` itself does not work with `unicode` when used standalone.
